subworkflows/local/longread_preprocessing.nf

```diff
@@ -52,29 +54,48 @@ workflow LONGREAD_PREPROCESSING {
         }
 
         if (!params.keep_lambda) {
-            NANOLYSE (
-                ch_long_reads,
-                ch_nanolyse_db
-            )
-            ch_long_reads = NANOLYSE.out.fastq
-            ch_versions = ch_versions.mix(NANOLYSE.out.versions.first())
+            if (params.longread_phageremoval_tool == 'chopper') {
+                CHOPPER (
```

Comment:
So theoretically CHOPPER can also be used for quality and length filtering , so maybe we should make this an option too?

It might need slightly more complicated conditions to know when both phageremvoal _and_ filtering is set to chopper, but it would certainly be more efficient too

Reply:
Yes, thought about fixing that. Coming up!